pkg/scyllaclient/model.go

```diff
@@ -239,3 +239,30 @@ func makeScyllaFeatures(endpointStates []*models.EndpointState) map[string]Scyll
 
 	return sfs
 }
+
+// ViewBuildStatus defines build status of a view.
+type ViewBuildStatus string
+
+// ViewBuildStatus enumeration.
+const (
+	StatusUnknown ViewBuildStatus = "UNKNOWN"
+	StatusStarted ViewBuildStatus = "STARTED"
+	StatusSuccess ViewBuildStatus = "SUCCESS"
+)
+
+// ViewBuildStatusOrder lists all view build statuses in the order of their execution.
+var ViewBuildStatusOrder = []ViewBuildStatus{
+	StatusUnknown,
+	StatusStarted,
+	StatusSuccess,
+}
+
+// Index returns status position in ViewBuildStatusOrder.
+func (s ViewBuildStatus) Index() int {
+	for i, stage := range ViewBuildStatusOrder {
+		if s == stage {
+			return i
+		}
+	}
+	panic("Unknown status: " + s)
+}
```

Comment:
It's equivalent to https://go.dev/play/p/wTMQowni_rX
And the Index() method is not needed then.

Reply:
Not really, as this implementation lacks a function to convert string into the enum type, which is needed when parsing Scylla response. So your suggestion requires 2 conversions (string -> int, int -> string) and the current one requires only 1 (string -> int).

I made this implementation mainly because that's how we handle string enums in SM (e.g. backup stage enum), so do you think that we should change the way in which we handle it in other places as well?

Reply:
> Not really, as this implementation lacks a function to convert string into the enum type,

The function below does the conversion.
By adding String() string function to the struct, you are defining the conversion to string.

If someone will do fmt.Println("%s", vbs)
Then the String() method is executed.

func (vbs ViewBuildStatus) String() string {
	switch vbs {
	case StatusUnknown:
		return "UNKNOWN"
	case StatusStarted:
		return "STARTED"
	case StatusSuccess:
		return "SUCCESS"
	}
	return ""
}

Reply:
>>Not really, as this implementation lacks a function to convert string into the enum type,

> The function below does the conversion.

But what about the conversion string -> int?
Scylla returns us a few raw strings, we want to compare them in order to choose the "earliest" one in terms of enumeration. How would you implement `ViewBuildStatus` with your implementation?

Reply:
> But what about the conversion string -> int?
Scylla returns us a few raw strings, we want to compare them in order to choose the "earliest" one in terms of enumeration. How would you implement ViewBuildStatus with your implementation?

Take a look at the example once again: https://go.dev/play/p/DuNMrNa06tk
ViewBuildStatus is of int type
`type ViewBuildStatus int`
So you can compare it with int explicitly.

By adding String() string method to any struct, you are instructing compiler how to convert the value to string.
fmt.Println(var) expects string. If ViewBuildStatus would not have String() method then fmt.Println(var) does something like fmt.Println(fmt.Sprintf("%d", var)), and the int value is printed.

Reply:
The String() method is int -> string conversion.
But the input to this problem are string and not ints.
How would you fill the following code fragment with your implementation:
```go

type ViewBuildStatus int

// ViewBuildStatus enumeration (keeps the order)
const (
	StatusUnknown ViewBuildStatus = iota
	StatusStarted
	StatusSuccess
)

func (vbs ViewBuildStatus) String() string {
	switch vbs {
	case StatusUnknown:
		return "UNKNOWN"
	case StatusStarted:
		return "STARTED"
	case StatusSuccess:
		return "SUCCESS"
	}
	return ""
}

func main() {
  s1 := "STARTED"
  s2 := "SUCCESS"
  
  // compare statuses and print the "earliest" one
}
```

Reply:
Like that https://go.dev/play/p/rU3JPkiUpHU
There is a constructor added.

But I actually see now that it's just a syntax sugar...
I'm fine with your proposal, but I don't like the fact that Index() method iterates over the slice.

Reply:
Please take a look at the commit `utils/slice: introduce Index` and tell me if that looks better to you. It still relies on a slice, but at least it's "immutable" and we have generic function for stage/status enumeration.